ci/lib.sh

```diff
@@ -182,6 +185,9 @@ osx-clang|osx-gcc)
 	if [ "$jobname" = osx-gcc ]
 	then
 		export CC=gcc-9
+		MAKEFLAGS="$MAKEFLAGS PYTHON_PATH=$(which python3)"
+	else
+		MAKEFLAGS="$MAKEFLAGS PYTHON_PATH=$(which python2)"
```

Comment:
The only objection I have is that it is already pretty magic to anybody who was not involved in the development of our CI/PR pipeline _where_ and _how_ the different axes are tested. This exacerbates that problem... How are people to know that the `-gcc` axis is intended to test with Python 3.x whereas the `-clang` ones are intended to test with Python 2.x?